src/main/java/com/couchbase/lite/replicator/ChangeTracker.java

```diff
@@ -630,10 +644,11 @@ public void setPaused(boolean paused) {
 
     protected void waitIfPaused(){
         synchronized (pausedObj) {
-            while (paused) {
+            while (paused && running) {
```

Comment:
This could have race condition on `running` as it doesn't get synchronized with pausedObj.